### PR TITLE
fix #1229 disable installer prompts via do not show again option

### LINKED ISSUE #1229: Be able to disable pylint globally from alert banner
When you open a new workspace that doesn't have pylint installed, the extension prompts you to either install pylint, disable the prompt, or simply close it. Unfortunately the disablement of pylint is local to the workspace and not globally. It would be great to have a way to turn off the prompting globally in your settings. Bonus points if the global disablement still allows the usage of pylint if it happens to be installed later (i.e. just disable the banner, not the ability to use the pylint integration).

/cc @lmazuel

### PR DESCRIPTION
**Pull request by DonJayamanne**
fix #1229

### CODE REVIEW

File: src/client/common/configSettings.ts [hunk 1]
```diff
@@ -166,6 +167,8 @@ export class PythonSettings extends EventEmitter implements IPythonSettings {
         this.devOptions = systemVariables.resolveAny(pythonSettings.get<any[]>('devOptions'))!;
         this.devOptions = Array.isArray(this.devOptions) ? this.devOptions : [];
         let lintingSettings = systemVariables.resolveAny(pythonSettings.get<ILintingSettings>('linting'))!;
+        this.disablePromptForFeatures = pythonSettings.get<string[]>('disablePromptForFeatures')!;
+        this.disablePromptForFeatures = Array.isArray(this.disablePromptForFeatures) ? this.disablePromptForFeatures : [];;
```
Comment (brettcannon) [inline src/client/common/configSettings.ts]: What is the double `;;` for?

File: src/client/common/installer.ts [hunk 2]
```diff
@@ -146,9 +143,9 @@ ProductTypes.set(Product.yapf, ProductType.Formatter);
 ProductTypes.set(Product.rope, ProductType.RefactoringLibrary);
 
 export class Installer implements vscode.Disposable {
-    private static terminal: vscode.Terminal;
+    private static terminal: vscode.Terminal | undefined | null;
```
Comment (brettcannon) [inline src/client/common/installer.ts]: Out of curiosity, what does this pattern do?

File: src/client/common/installer.ts [hunk 3]
```diff
@@ -158,46 +155,65 @@ export class Installer implements vscode.Disposable {
...
+            else {
+                console.warn(message);
+            }
+            return;
+        }
 
         const installOption = 'Install ' + productName;
         const disableOption = 'Disable ' + productTypeName;
-        const disableOptionGlobally = `Disable ${productTypeName} globally`;
+        const dontShowAgain = `Don't show this again`;
```
Comment (brettcannon) [inline src/client/common/installer.ts]: "Don't show this prompt again"?

File: src/client/common/installer.ts [hunk 4]
```diff
@@ -275,12 +291,12 @@ export function disableLinter(product: Product, global?: boolean) {
     }
 }
 
-function isProductInstalled(product: Product): Promise<boolean | undefined> {
+async function isProductInstalled(product: Product): Promise<boolean | undefined> {
```
Comment (brettcannon) [inline src/client/common/installer.ts]: To help me learn, what are the rules for when to use `null` versus `undefined`?

File: src/client/common/installer.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/common/installer.ts]: Allow it to be nullable (i'm slowly introducing some strict checks.
I don't want to turn this on globally, else we'll have 100s of errors.
[New --strict master option](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html)

File: src/client/common/installer.ts (hunk 4 above)
Comment (DonJayamanne) [reply src/client/common/installer.ts]: lol, ```undefined``` is the similar to ```None``` in Python, ```null``` is more or less an actual value.
Generally in JS people use either, as both evaluate to nothing.
Don't you love JS.

File: src/client/common/configSettings.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/common/configSettings.ts]: woops

Comment (DonJayamanne) [reply src/client/common/configSettings.ts]: Fixed.

File: src/client/common/installer.ts (hunk 4 above)
Comment (brettcannon) [reply src/client/common/installer.ts]: Figures. Wasn't sure if there was a move towards just `null` or something to overlook the weirdness of having both values or to use the `!` operator or something.

File: src/client/common/installer.ts (hunk 2 above)
Comment (brettcannon) [reply src/client/common/installer.ts]: Is the errors from dependencies or just old code that needs an update?

File: src/client/common/installer.ts (hunk 4 above)
Comment (DonJayamanne) [reply src/client/common/installer.ts]: e.g. if you declare a variable or call a function without passing in a value, the default value of those variables or parameters is ```undefined```.
```null``` is a value that you'd assign manually to something.

**Pull request merged**